File: self/S2_07052025/ConvertPython/Ex2_Python/Ex2_6.py
import cv2
import numpy as np
import matplotlib.pyplot as plt

live = cv2.imread('live.jpg') / 255.0
mask = cv2.imread('mask.jpg') / 255.0
dif_img = np.sqrt(np.abs(live - mask))
plt.figure(figsize=(12, 4))
plt.subplot(1, 3, 1), plt.imshow(live), plt.title('Live Image', fontsize=15), plt.axis('off')
plt.subplot(1, 3, 2), plt.imshow(mask), plt.title('Mask Image', fontsize=15), plt.axis('off')
plt.subplot(1, 3, 3), plt.imshow(dif_img), plt.title('Different Image', fontsize=15), plt.axis('off')
plt.show()
plt.savefig('live_mask.jpg')

File: self/S2_07052025/Ex2_6.m
% Subtracting two image
clear, clc, close all;
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
pkg load image; %load image package in Octave

%Load test image
live = im2double(imread('live.jpg'));
mask = im2double(imread('mask.jpg'));

dif_Img = imsubtract(live, mask) .^ 0.5;

%plot image
FS = 15
figure, clf;

subplot(1, 3, 1), imshow(live, [0, 1]), title('Live Image', "fontsize", FS);
subplot(1, 3, 2), imshow(mask, [0, 1]), title('Mask Image', "fontsize", FS);
subplot(1, 3, 3), imshow(dif_Img, [0, 1]), title('Different Image', "fontsize", FS);

saveas(gcf, 'live_mask.jpg');

